src/storage/index/index_manager.cpp

```diff
@@ -0,0 +1,193 @@
+#include "storage/index/index_manager.h"
+#include <exception>
+#include <memory>
+#include <string>
+#include <vector>
+#include "catalog/index_handle.h"
+#include "util/transaction_benchmark_util.h"
+
+namespace terrier::storage::index {
+
+Index *IndexManager::GetEmptyIndex(catalog::index_oid_t index_oid, SqlTable *sql_table, bool unique_index,
+                                   const std::vector<std::string> &key_attrs) {
+  // Setup the oid and constraint type for the index
+  IndexFactory index_factory;
+  ConstraintType constraint = (unique_index) ? ConstraintType::UNIQUE : ConstraintType::DEFAULT;
+  index_factory.SetOid(index_oid);
+  index_factory.SetConstraintType(constraint);
+
+  // Setup the key schema for the index
```

Comment:
This logic assumes that you're indexing on columns, but you can build an index on expressions as well. It's probably too late to change this implementation now, but I wanted to point it out.

Reply:
Do we have infrastructure to evaluate expressions now? We can definitely implement this if time permits.